modules/adfs/lib/IdP/ADFS.php

```diff
@@ -70,7 +71,13 @@ public static function ADFS_GenerateResponse($issuer, $target, $nameid, $attribu
 		foreach ($attributes as $name => $values) {
 			if ((!is_array($values)) || (count($values) == 0)) continue;
 			$hasValue = FALSE;
-			$r = '<saml:Attribute AttributeNamespace="http://schemas.xmlsoap.org/claims" AttributeName="' . htmlspecialchars($name) .'">';
+			if (strpos($name, '/')) {
```

Comment:
If it contains a '/' it will assume the attribute name is with namespace... Checking if it is in fact a namespace contained attribute-name seems a challenge... Worst case: a invalid namespace is issued, which simply doesn't work...